src/python/vehicle.py

```diff
@@ -0,0 +1,183 @@
+            loss1 = translateJSONtoLOSSfunction(resources, contents[0])
+            loss2 = translateJSONtoLOSSfunction(resources, contents[1])
+
+            def result_func(context):
+                return min(loss1(context), loss2(context))
+
+            return result_func
+
+        elif tag == 'Max':
+            loss1 = translateJSONtoLOSSfunction(resources, contents[0])
+            loss2 = translateJSONtoLOSSfunction(resources, contents[1])
+
+            def result_func(context):
+                return max(loss1(context), loss2(context))
+
+            return result_func
+
+        elif tag == 'Sub':
+            loss1 = translateJSONtoLOSSfunction(resources, contents[0])
+            loss2 = translateJSONtoLOSSfunction(resources, contents[1])
+
+            def result_func(context):
+                return loss1(context) - loss2(context)
+
+            return result_func
+
+        elif tag == 'Ind':
+            loss1 = translateJSONtoLOSSfunction(resources, contents[0])
+            loss2 = translateJSONtoLOSSfunction(resources, contents[1])
+
+            def result_func(context):
+                if loss1(context) == loss2(context):
+                    return 1
+                else:
+                    return 0
+
+            return result_func
+
+        elif tag == 'At':
+            lossList = contents[0]
+            index = contents[1]
+            lossIndex = translateJSONtoLOSSfunction(resources, index)
+            lossTensor = translateJSONtoLOSSfunction(resources, lossList)
+
+            def result_func(context):
+                return lossTensor(context)[lossIndex(context)]
+
+            return result_func
+            
+        elif tag == 'NetApp':      
+            model = resources[contents[0]]
+            lossInputsList = []
+            for c in contents[1]:
+                lossInputsList.append(translateJSONtoLOSSfunction(resources, c))
+
+            def result_func(context):
+                inputs = []
+                for l in lossInputsList:
+                    inputs.append(l(context))
+                output = model(inputs, training=True)
+                return output
+
+            return result_func
+
+        elif tag == 'Quant':
+            body = contents[3]
+            bodyloss = translateJSONtoLOSSfunction(resources, body)
+            domainloss = translateJSONtoLOSSfunction(resources, contents[2])
+
+            if contents[0] == 'All':
+                def result_func(context):
+                    domain = domainloss(context)
+                    maxLoss = np.NINF
+                    for _ in range(10):
+                        context.insert(0, random.uniform(domain[0], domain[1]))
+                        value = bodyloss(context)
+                        if value > maxLoss:
+                            maxLoss = value
+                        context.pop(0)
+                    return maxLoss
+
+            elif contents[0] == 'Any':
+                def result_func(context):
```

Comment:
This function is identical to the one above, apart from using `min` rather than `max`. Maybe simply have one function and pass in either the `min` or `max` operation as an argument?

Reply:
I could merge the two like this:
`def result_func(context):
      domain = domainloss(context)
      minLoss = np.Inf
      maxLoss = np.NINF
      for _ in range(10):
          context.insert(0, random.uniform(domain[0], domain[1]))
          value = bodyloss(context)
          if contents[0] == 'Any' and value < minLoss:
              minLoss = value
          elif contents[0] == 'All' and value > maxLoss:
              maxLoss = value
          context.pop(0)
      return minLoss`
Do you prefer it?

Reply:
Yup, that seems better!

Reply:
Also on Github you can use ``` <newline> code <newline> ``` to type set blocks of code. And also best not to resolve comments before the problem is solved :smile: